roles/custom/matrix-synapse/templates/synapse/labels.j2

```diff
@@ -82,7 +82,7 @@ traefik.http.routers.matrix-synapse-public-client-api.tls.certResolver={{ matrix
 
 
 
-{% if matrix_synapse_container_labels_internal_client_api_enabled %}
+{% if matrix_synapse_container_labels_internal_client_api_enabled and not matrix_synapse_reverse_proxy_companion_enabled %}
```

Comment:
The check was added here instead of `roles/custom/matrix-synapse/defaults/main.yml` since `matrix_synapse_container_labels_internal_client_api_enabled` is configured in `group_vars/matrix_servers` depending on the proxy setup being used.